### PR TITLE
Fixed bug which made the top level inferred schema optional

### PR DESCRIPTION
**Pull request by rozza**
KAFKA-171

### CODE REVIEW

File: src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java [hunk 1]
```diff
@@ -36,6 +36,19 @@
   private static final Schema DEFAULT_INFER_SCHEMA_TYPE = Schema.OPTIONAL_STRING_SCHEMA;
   public static final String SCHEMA_NAME_TEMPLATE = "inferred_name_%s";
 
+  public static Schema inferDocumentSchema(final BsonDocument document) {
+    SchemaBuilder builder = SchemaBuilder.struct();
+    if (document.containsKey(ID_FIELD)) {
```
Comment (jyemin) [inline src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java:41]: Why does _id have to go first?

File: src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java [hunk 2]
```diff
@@ -36,6 +36,19 @@
   private static final Schema DEFAULT_INFER_SCHEMA_TYPE = Schema.OPTIONAL_STRING_SCHEMA;
   public static final String SCHEMA_NAME_TEMPLATE = "inferred_name_%s";
 
+  public static Schema inferDocumentSchema(final BsonDocument document) {
+    SchemaBuilder builder = SchemaBuilder.struct();
```
Comment (jyemin) [inline src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java:40]: Can we DRY this up with the code in inferSchema?

**Review by jyemin — CHANGES_REQUESTED** [review]

File: src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java (hunk 1 above)
Comment (rozza) [reply src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java:41]: It doesn't have to but we follow the idom that the _id field is always the first field in the documents.

File: src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java (hunk 2 above)
Comment (rozza) [reply src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java:40]: I had hoped to but  it ended up making the code more complex and harder to read. I'll take another pass.

Comment (rozza) [reply src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java:40]: Well  that was easy now the document schema inference was already in its own method.

**Review by durran — APPROVED** [review]
LGTM

**Review by jyemin — APPROVED** [review]
LGTM

**Pull request merged**